contracts/utils/cryptography/MerkleProof.sol

```diff
@@ -23,21 +23,36 @@ library MerkleProof {
         bytes32 root,
         bytes32 leaf
     ) internal pure returns (bool) {
-        bytes32 computedHash = leaf;
+        // Check if the computed hash (root) is equal to the provided root
+        (bytes32 computedHash, ) = processProof(proof, leaf);
+        return computedHash == root;
+    }
 
+    /**
+     * @dev Returns the rebuild hash of node obtained by traversing a Merklee
+     * tree up from `leaf` using `proof`. If `proof` proof is valid, in the send
+     * that it contains the hashes of the sibling on the branch from the leaf to
+     * the root, then this will return the hash of the root of the tree. In
```

Comment:
```suggestion
     * @dev Returns the rebuilt hash of nodes obtained by traversing a Merklee
     * tree up from `leaf` using `proof`. If `proof` is valid, in the sense
     * that it contains the hashes of the siblings on the branch from the leaf to
     * the root, then this will return the hash of the root of the tree. In
```